### PR TITLE
Fix perf issue for auto trait selection

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Follow-up of https://github.com/rust-lang/rust/pull/95069 which fixes the perf issue introduced by it.

r? @oli-obk

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/select/mod.rs [hunk 1]
```diff
@@ -1272,7 +1272,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         // the master cache. Since coherence executes pretty quickly,
         // it's not worth going to more trouble to increase the
         // hit-rate, I don't think.
-        if self.intercrate || self.allow_negative_impls {
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/traits/select/mod.rs:1275]: Please remove the allow_negative_impls field entirely

**Comment by GuillaumeGomez** [comment]
Let's run another perf check to see if there is any difference with one field removed.

@bors try @rust-timer queue

**Pull request merged**